src/util.js

```diff
@@ -108,10 +108,32 @@ exports.deserialize = (data, callback) => {
   setImmediate(() => callback(null, deserialized))
 }
 
-exports.cid = (dagNode, callback) => {
+/**
+ * @callback CidCallback
+ * @param {?Error} error - Error if getting the CID failed
+ * @param {?CID} cid - CID if call was successful
+ */
+/**
+ * Get the CID of the DAG-Node.
+ *
+ * @param {Object} dagNode - Internal representation
+ * @param {Object} [options] - Options to create the CID
+ * @param {number} [options.version=1] - CID version number
+ * @param {string} [options.hashAlg] - Defaults to hashAlg for the resolver
+ * @param {CidCallback} callback - Callback that handles the return value
+ * @returns {void}
+ */
+exports.cid = (dagNode, options, callback) => {
+  if (options instanceof Function) {
+    callback = options
+    options = {}
+  }
+  options = options || {}
+  const hashAlg = options.hashAlg || resolver.defaultHashAlg
+  const version = options.version || 1
```

Comment:
This doesn't work correctly. If `options.version` is set to `0`, then `version` would become `1` as `0` evaluates to false.